File: intrinsic/skills/testing/no_op_skill.cc
// Copyright 2023 Intrinsic Innovation LLC

#include "intrinsic/skills/testing/no_op_skill.h"

#include <memory>

#include "absl/status/statusor.h"
#include "google/protobuf/descriptor.h"
#include "google/protobuf/message.h"
#include "intrinsic/skills/cc/skill_interface.h"
#include "intrinsic/skills/cc/skill_interface_utils.h"
#include "intrinsic/skills/proto/equipment.pb.h"
#include "intrinsic/skills/proto/skill_service.pb.h"
#include "intrinsic/skills/testing/no_op_skill.pb.h"

namespace intrinsic::skills {

std::unique_ptr<SkillInterface> NoOpSkill::CreateSkill() {
  return std::make_unique<NoOpSkill>();
}

absl::StatusOr<std::unique_ptr<google::protobuf::Message>> NoOpSkill::Execute(
    const ExecuteRequest& request, ExecuteContext& context) {
  return nullptr;
}

absl::StatusOr<std::unique_ptr<::google::protobuf::Message>> NoOpSkill::Preview(
    const PreviewRequest& request, PreviewContext& context) {
  return PreviewViaExecute(*this, request, context);
}

}  // namespace intrinsic::skills

File: intrinsic/skills/testing/no_op_skill.py
# Copyright 2023 Intrinsic Innovation LLC
# Intrinsic Proprietary and Confidential
# Provided subject to written agreement between the parties.

"""A skill that always returns OK and has an empty equipment set."""

from intrinsic.skills.proto import footprint_pb2
from intrinsic.skills.python import skill_interface as skl
from intrinsic.skills.testing import no_op_skill_pb2
from intrinsic.util.decorators import overrides


class NoOpSkill(skl.Skill):
  """Skill that always returns OK and has an empty equipment set."""

  @overrides(skl.Skill)
  def execute(
      self,
      request: skl.ExecuteRequest[no_op_skill_pb2.NoOpSkillParams],
      context: skl.ExecuteContext,
  ) -> None:
    """See base class."""
    return None

  @overrides(skl.Skill)
  def get_footprint(
      self,
      params: skl.GetFootprintRequest[no_op_skill_pb2.NoOpSkillParams],
      context: skl.GetFootprintContext,
  ) -> footprint_pb2.Footprint:
    """See base class."""
    return footprint_pb2.Footprint()
